github.opam

```diff
@@ -37,7 +37,7 @@ depends: [
   "cohttp" {>= "0.99.0" & < "3.0.0"}
   "cohttp-lwt" {>= "0.99.0" & < "3.0.0"}
   "lwt" {>= "2.4.4"}
-  "github-data" {= version}
+  "github_data" {= version}
```

Comment:
This is definitely wrong -- the opam package is `github-data`. Need to look into why opam-dune-lint is suggesting this.

Reply:
The message from opam-dune-lint is:
```
WARNING: can't find opam package providing "github_data"!
github-data.opam: OK
github-jsoo.opam: OK
github-unix.opam: OK
github.opam: changes needed:
  "github_data" {>= 0}                     [from lib]
```

The problem is that `lib` is (incorrectly) referring to `github_data`.

Reply:
Thanks for the pointer, reduced this change to the minimal fix.